pkg/alertmanager/operator.go

```diff
@@ -644,6 +644,11 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 		return nil
 	}
 
+	// Check if the Alertmanager instance is marked for deletion.
```

Comment:
I'd suggest to get inspiration from the UpdateStatus() method and do the check at the very beginning of the function.

https://github.com/prometheus-operator/prometheus-operator/blob/535cd8cf87670b812a12b4fb500ec92d06689877/pkg/alertmanager/operator.go#L782-L789

Reply:
Thanks for your comment. I've added thanos and the agent, reused the DeletionInProgress function and moved the checks as high as possible.